flaml/model.py

```diff
@@ -2099,6 +2099,129 @@ class XGBoostLimitDepth_TS(TS_SKLearn):
     base_class = XGBoostLimitDepthEstimator
 
 
+class MultiModalEstimator(BaseEstimator):
+    """
+    The class for tuning AutoGluon TextPredictor
+    """
+    @classmethod
+    def search_space(cls, **params):
+        """
+        Add the possible search space configs here, e.g. 'optimization.lr'
+        reference:
+        https://auto.gluon.ai/stable/tutorials/text_prediction/customization.html#custom-hyperparameter-values
+        """
+        # TODO: expand the search space
+        search_space_dict = {
+            "model.network.agg_net.mid_units": {
+                "domain": tune.choice(list(range(32, 129))),
+                "init_value": 128,
+            },
+            "optimization.lr": {
+                "domain": tune.loguniform(lower=1E-5, upper=1E-4),
+                "init_value": 1E-4,
+            },
+            "optimization.wd": {
+                "domain": tune.choice([1E-4, 1E-3, 1E-2]),
+                "init_value": 1E-4,
+            },
+            "optimization.warmup_portion": {
+                "domain": tune.choice([0.1, 0.2]),
+                "init_value": 0.1, 
+            },
+            "optimization.layerwise_lr_decay": {
+                "domain": tune.choice([0.8, 0.9]),
+                "init_value": 0.8,
+            },
+            "optimization.nbest": {
+                "domain": tune.choice([2, 3, 4,]),
+                "init_value": 3,
+            },
+            "optimization.num_train_epochs": {
+                "domain": tune.choice([5, 10, 15,]),
+                "init_value": 10,
+            },
+            "optimization.per_device_batch_size": {
+                "domain": tune.choice([2, 4, 8,]),
+                "init_value": 10,
+            },
+            "optimization.batch_size": {
+                "domain": tune.choice([32, 64, 128,]),
+                "init_value": 128,
+            },
+        }
+        return search_space_dict

```

Comment:
There were only 4 hyperparameters and now there are 9. Which one was the search space used in your original experiment for autogluon?

Reply:
The original four are "model.network.agg_net.mid_units", "optimization.warmup_portion", "optimization.lr", "optimization.wd".